OCPP-J/src/main/java/eu/chargetime/ocpp/WebSocketListener.java

```diff
@@ -161,6 +166,17 @@ public void onStart() {
 
     configure();
     server.start();
+    try {
+        socketBound.get(SOCKET_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+    } catch (ExecutionException ex) {
+        close();
+        throw (SocketException) ex.getCause();
```

Comment:
suggestion(non-blocking): I know that the only exception being passed is a `SocketException` since it is checked with `instanceof SocketException`, but this would be more future-proof if it checked the exception here before casting it.

Perhaps the `instanceof` check should entirely be moved here, but I haven't checked what other errors `onError()` might produce.